### PR TITLE
fast forward doc to web3mail v0.6.0

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
doc update for https://github.com/iExecBlockchainComputing/web3mail-sdk/pull/120
- add all missing methods and options to refect web3mail 0.6.0 interface
- format web3mail code block for consistency

Misc
- rework required/optional parameters => `(required)` no longer specifier in titles to avoid displaying in menu
- prettier

### CODE REVIEW

File: documentation/tools/web3mail/advanced-configuration.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Advanced configuration
+
+The `IExecWeb3mail` constructor accepts configuration options object.
+As these options are very specific you wont need use them for a standard usage of `@iexec/web3mail`.
```
Comment (cedric25) [inline documentation/tools/web3mail/advanced-configuration.md:4]:
```suggestion
As these options are very specific, you won't need to use them for a standard usage of `@iexec/web3mail`.
```

File: documentation/tools/web3mail/advanced-configuration.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+new IExecWeb3mail(ethProvider, options);
+```
+
+## Options
+
+### dappAddressOrENS
+
+The Ethereum contract address or ENS (Ethereum Name Service) for the web3mail dapp.
+
+If not provided, the default ENS `web3mail.apps.iexec.eth` pointing to the latest version of the web3mail provided by iExec will be used.
```
Comment (cedric25) [inline documentation/tools/web3mail/advanced-configuration.md:16]:
```suggestion
If not provided, the default ENS `web3mail.apps.iexec.eth` pointing to the latest version of the web3mail dapp provided by iExec will be used.
```

File: documentation/tools/web3mail/advanced-configuration.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+```js
+new IExecWeb3mail(ethProvider, {
+  dappAddressOrENS: "web3mail.apps.iexec.eth",
+});
+```
+
+### dappWhitelistAddress
+
+The Ethereum contract address for the web3mail dapps whitelist contract. By granting access to a whitelist, email address owner ensure their email is still available for consumers when a new version of web3mail dapp is released.
```
Comment (cedric25) [inline documentation/tools/web3mail/advanced-configuration.md:26]:
```suggestion
The Ethereum contract address for the web3mail dapps whitelist contract. By granting access to a whitelist, email address owners ensure their email is still available to consumers even after a new version of web3mail dapp gets released.
```

File: documentation/tools/web3mail/methods/fetchusercontacts.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# fetchUserContacts
+
+Method to get the ETH addresses that gave access to their protected data for messaging usage to a specific user address or ENS.
+
+## Usage
+
+```javascript
+const listContact = await web3mail.fetchUserContacts(
```
Comment (cedric25) [inline documentation/tools/web3mail/methods/fetchusercontacts.md:8]:
```suggestion
const contactsList = await web3mail.fetchUserContacts(
```

File: documentation/tools/web3mail/methods/sendemail.md [hunk 5]
```diff
@@ -23,49 +21,128 @@ const sendEmail = await web3mail.sendEmail({
...
+    protectedData: "0xA0Cf798816D4b9b9866b5330EEa46a18382f251e",
+    emailSubject: "My email subject",
+    emailContent: "My email content",
+<strong>    senderName: "Awesome project team",
+</strong>})
+</code></pre>
+
+### label
+
+Allow adding a custom public label that will be written onchain as `iexec_args` in the deal params.
```
Comment (cedric25) [inline documentation/tools/web3mail/methods/sendemail.md:91]:
```suggestion
Allows adding a custom public label that will be written onchain as `iexec_args` in the deal params.
```

File: documentation/tools/web3mail/methods/fetchusercontacts.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+Method to get the ETH addresses that gave access to their protected data for messaging usage to a specific user address or ENS.
+
+## Usage
+
+```javascript
+const contactsList = await web3mail.fetchUserContacts(
+  "0xF048eF3d7E3B33A465E0599E641BB29421f7Df92"
+);
+```
+
```
Comment (Le-Caignec) [inline documentation/tools/web3mail/methods/fetchusercontacts.md:12]: Add

```
## Parameter

### userAddress

Specify the Ethereum address of the user from whom you wish to obtain the list of contact.

<pre class="language-javascript"><code class="lang-javascript">const contactsList = await web3mail.fetchUserContacts({
<strong>userAddress: "0xF048eF3d7E3B33A465E0599E641BB29421f7Df92"
</strong> });
</code></pre>
```

**Review by Le-Caignec — APPROVED** [review]
I appreciated the inclusion of `optional` keywords specified within parentheses. I didn't find it particularly helpful & necessary to title sections as 'Optional Parameters' since there is already a 'Parameters' title above.

File: documentation/tools/web3mail/methods/fetchusercontacts.md (hunk 6 above)
Comment (PierreJeanjacquot) [reply documentation/tools/web3mail/methods/fetchusercontacts.md:12]: added

File: documentation/tools/web3mail/methods/sendemail.md [hunk 7]
```diff
@@ -23,49 +21,128 @@ const sendEmail = await web3mail.sendEmail({
...
-    emailSubject: 'My email subject',
-<strong>    emailContent: 'My mail content'
+    protectedData: "0xA0Cf798816D4b9b9866b5330EEa46a18382f251e",
+    emailSubject: "My email subject",
+<strong>    emailContent: "My email content",
 </strong>})
 </code></pre>
 
-### contentType (optional)
+## Optional parameters
```
Comment (Le-Caignec) [inline documentation/tools/web3mail/methods/sendemail.md:63]: you miss this one

**Review by Le-Caignec — APPROVED** [review]

**Review by Le-Caignec — APPROVED** [review]

**Pull request merged**